### PR TITLE
CSPL-1712: Enterprise common API changes

### PR DESCRIPTION
**Pull request by akondur**
Move Spec from `pkg/splunk/common/common_types.go` to `api/v3/common_types.go`. The `Spec` struct was sitting in the `pkg/splunk/common` directory while it was being used to build the CRD spec. 

The change is to maintain uniformity i.e keep all the CRD spec generating code in the `api/` package.

Moving forward:
- `api/` package can be imported by package inside `pkg/splunk` 
- `pkg/splunk/common` can be imported by all other packages inside `pkg/splunk` directory i.e `pkg/splunk/client, pkg/splunk/controller, pkg/splunk/enterprise, pkg/splunk/util, pkg/splunk/test`. 
- The `api/` package cannot import `pkg/splunk/common`(which was the case before this change) to avoid cyclic dependencies and keep the `api` package as only importable and restrict its access to import any package in the `pkg/splunk` directory.

Other minor changes:
- Changed `enterprisev3` to `enterpriseApi` to ensure uniformity
- Corrected typo - `%s/standlaone/standalone`
- Removed the comment with `operator-sdk generate k8s` as its no longer relevant

### CODE REVIEW

**Review by vivekr-splunk — APPROVED** [review]

File: pkg/splunk/common/util.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ import (
 	"strings"
 	"time"
 
+	enterpriseApi "github.com/splunk/splunk-operator/api/v3"
```
[the remark this answers is not in the document]
Comment (akondur) [reply pkg/splunk/common/util.go:29]: Yes, `common` has a dependency of `enterpriseApi`. `enterpriseApi` is just a set of structs which form our CRDs using basic golang data structures or APIs from K8S such as `corev1`, `metav1` etc... It does not need to refer to anything Splunk specific from `splunk/common`. If you refer the directory structure the [api directory](https://github.com/splunk/splunk-operator/tree/develop/api) is outside of the [pkg/splunk/ directory](https://github.com/splunk/splunk-operator/tree/develop/pkg/splunk) to make the distinction. `enterpriseApi` can be used by any package as every package should have access to the CRD structs.

File: pkg/splunk/common/util.go [hunk 2]
```diff
@@ -367,7 +368,7 @@ func ValidateResources(resources *corev1.ResourceRequirements, defaults corev1.R
 }
 
 // ValidateSpec checks validity and makes default updates to a Spec, and returns error if something is wrong.
-func ValidateSpec(spec *Spec, defaultResources corev1.ResourceRequirements) error {
+func ValidateSpec(spec *enterpriseApi.Spec, defaultResources corev1.ResourceRequirements) error {
```
[the remark this answers is not in the document]
Comment (akondur) [reply pkg/splunk/common/util.go:371]: The `enterpriseApi` directory is a bunch of structs forming CRDs. We do not have reconcile logic linked there to validate spec. If you look at the current validations , we validate `CommonSplunkSpec` in the splunk/enterprise directory [here](https://github.com/splunk/splunk-operator/blob/a2eb64aee200434700f88d67364133d331f035d4/pkg/splunk/enterprise/configuration.go#L196).

File: pkg/splunk/common/util.go [hunk 3]
```diff
@@ -383,7 +384,7 @@ func ValidateSpec(spec *Spec, defaultResources corev1.ResourceRequirements) erro
 }
 
 // setServiceTemplateDefaults sets default values for service templates
-func setServiceTemplateDefaults(spec *Spec) {
+func setServiceTemplateDefaults(spec *enterpriseApi.Spec) {
```
[the remark this answers is not in the document]
Comment (akondur) [reply pkg/splunk/common/util.go:387]: Same answer as above. An alternative could be we move this to `splunk/enterprise` to maintain consistency?

File: pkg/splunk/common/util_test.go [hunk 4]
```diff
@@ -24,6 +24,7 @@ import (
 	"sync"
 	"testing"
 
+	enterpriseApi "github.com/splunk/splunk-operator/api/v3"
```
[the remark this answers is not in the document]
Comment (akondur) [reply pkg/splunk/common/util_test.go:27]: Same answer as above. No testing logic in `enterpriseApi`

File: pkg/splunk/common/util.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jryb) [reply pkg/splunk/common/util.go:371]: I guess what I'm saying is this isn't validating any struct here in the common/util.go anymore and sincew the struct moved, this validate code should move too.

File: pkg/splunk/common/util.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (jryb) [reply pkg/splunk/common/util.go:387]: Ya thats what I was thinking.  Move it to where the struct lives now with the other setters and validaters for this code.

File: pkg/splunk/common/util.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jryb) [reply pkg/splunk/common/util.go:29]: I think if you move the validate/set functions for CommonSpec below then you can get rid of this include.  This should keep it more modular.

**Review by jryb — APPROVED** [review]

**Review by sgontla — APPROVED** [review]

**Review by kashok-splunk — APPROVED** [review]

File: pkg/splunk/enterprise/clustermaster.go [hunk 5]
```diff
@@ -258,7 +258,7 @@ func getClusterManagerStatefulSet(ctx context.Context, client splcommon.Controll
 	smartStoreConfigMap := getSmartstoreConfigMap(ctx, client, cr, SplunkClusterManager)
 
 	if smartStoreConfigMap != nil {
-		setupInitContainer(&ss.Spec.Template, cr.Spec.Image, cr.Spec.ImagePullPolicy, commandForCMSmartstore)
```
Comment (akondur) [inline pkg/splunk/enterprise/clustermaster.go:261]: Bug: Needs fix

**Pull request merged**